examples/cmri_mpca/main.py

```diff
@@ -70,43 +70,69 @@ def main():
     y = landmark_df["Group"].values
     y[np.where(y != 0)] = 1  # convert to binary classification problem, i.e. no PH vs PAH
 
-    # plot the first phase of images
+    # plot the first phase of images with landmarks
+    marker_names = list(landmark_df.columns[1::2])
+    markers = []
+    for marker in marker_names:
+        marker_name = marker.split(" ")
+        marker_name.pop(-1)
+        marker_name = " ".join(marker_name)
+        markers.append(marker_name)
+
     if save_images:
-        visualize.plot_multi_images(
-            [images[i][0, ...] for i in range(n_samples)],
-            marker_locs=landmarks,
-            im_kwargs=dict(cfg.IM_KWARGS),
-            marker_kwargs=dict(cfg.MARKER_KWARGS),
-            n_cols=10,
-        ).savefig(str(save_images_location) + "/0)first_phase.png")
+        n_img_per_fig = 35
+        n_figures = int(n_samples / n_img_per_fig) + 1
+        for k in range(n_figures):
+            visualize.plot_multi_images(
+                [images[i][0, ...] for i in range(k * n_img_per_fig, min((k + 1) * n_img_per_fig, n_samples))],
+                marker_locs=landmarks[k * n_img_per_fig : min((k + 1) * n_img_per_fig, n_samples), :],
+                im_kwargs=dict(cfg.IM_KWARGS),
+                marker_cmap="Set1",
+                marker_kwargs=dict(cfg.MARKER_KWARGS),
+                marker_titles=markers,
+                image_titles=list(patient_ids[k * n_img_per_fig : min((k + 1) * n_img_per_fig, n_samples)]),
+                n_cols=5,
+            ).savefig(str(save_images_location) + "/0)landmark_visualization_%s_of_%s.png" % (k + 1, n_figures))
 
     # ---- data pre-processing ----
     # ----- image registration -----
     img_reg, max_dist = reg_img_stack(images.copy(), landmarks, landmarks[0])
     if save_images:
         visualize.plot_multi_images(
-            [img_reg[i][0, ...] for i in range(n_samples)], im_kwargs=dict(cfg.IM_KWARGS), n_cols=10
+            [img_reg[i][0, ...] for i in range(n_samples)],
+            im_kwargs=dict(cfg.IM_KWARGS),
+            n_cols=10,
+            image_titles=list(patient_ids),
         ).savefig(str(save_images_location) + "/1)image_registration")
 
     # ----- masking -----
     img_masked = mask_img_stack(img_reg.copy(), mask)
     if save_images:
         visualize.plot_multi_images(
-            [img_masked[i][0, ...] for i in range(n_samples)], im_kwargs=dict(cfg.IM_KWARGS), n_cols=10
+            [img_masked[i][0, ...] for i in range(n_samples)],
+            im_kwargs=dict(cfg.IM_KWARGS),
+            n_cols=10,
```

Comment:
If you are using the same para multiple times, config seems better in the long term.